pkg/datapath/linux/config/config.go

```diff
@@ -818,10 +822,6 @@ func (h *HeaderfileWriter) writeTemplateConfig(fw *bufio.Writer, e datapath.Endp
 		fmt.Fprintf(fw, "#define ENABLE_ROUTING 1\n")
 	}
 
-	if e.RequireEndpointRoute() {
```

Comment:
Didn't we try this at some point and establish that the (cilium-)CNI plugin must feed us this variable based on factors such as whether it is running in a chained setup, therefore we don't have the luxury of just swapping this for a simple `option.Config.EnableEndpointRoutes` check?

@pchaigno I recall you had worked on something similar in the past.

Reply:
Looking back through the history from the commit message links & PRs, I found that I similarly got confused last time between this setting and `RequireRouting()`: https://github.com/cilium/cilium/pull/17000#issuecomment-893822946

This should be fine on that side.

Reply:
Yes, this change is fine AFAIK. At least it's not worse than what we already do. The issue we had in the past were related to cleaning endpoint routes state when that setting is disabled (i.e., delete routes).